### PR TITLE
Migrate flutter_command to null safety

### PR DESCRIPTION
**Pull request by jmagman**
Migrate flutter_command and flutter_command_runner.

Part of #71511.

### CODE REVIEW

**Comment by Hixie** [comment]
test-exempt: code refactor with no semantic change.

File: packages/flutter_tools/lib/src/runner/flutter_command.dart [hunk 1]
```diff
@@ -1433,23 +1433,23 @@ abstract class FlutterCommand extends Command<void> {
...
     ].join('\n');
     return help;
   }
 
-  ApplicationPackageFactory applicationPackages;
+  ApplicationPackageFactory? applicationPackages;
 
   /// Gets the parsed command-line option named [name] as `bool`.
-  bool boolArg(String name) => argResults[name] as bool;
+  bool boolArg(String name) => argResults?[name] as bool? ?? false;
```
Comment (christopherfujino) [inline packages/flutter_tools/lib/src/runner/flutter_command.dart:1446]: Should we be defaulting to `false` rather than returning `null`?

Comment (jmagman) [reply packages/flutter_tools/lib/src/runner/flutter_command.dart:1446]: I think we want `false`, it's checked as a bool everywhere:
https://github.com/flutter/flutter/blob/9e88fe328efcf42d3037a965b39269f0805cd4ce/packages/flutter_tools/lib/src/commands/build_appbundle.dart#L89-L93
https://github.com/flutter/flutter/blob/9e88fe328efcf42d3037a965b39269f0805cd4ce/packages/flutter_tools/lib/src/commands/config.dart#L104
https://github.com/flutter/flutter/blob/9e88fe328efcf42d3037a965b39269f0805cd4ce/packages/flutter_tools/lib/src/commands/ide_config.dart#L161

Set here:
https://github.com/flutter/flutter/blob/9e88fe328efcf42d3037a965b39269f0805cd4ce/packages/flutter_tools/lib/src/commands/run.dart#L158
Checked as bool here:
https://github.com/flutter/flutter/blob/9e88fe328efcf42d3037a965b39269f0805cd4ce/packages/flutter_tools/lib/src/commands/run.dart#L448
etc

Comment (christopherfujino) [reply packages/flutter_tools/lib/src/runner/flutter_command.dart:1446]: makes sense

**Review by christopherfujino — APPROVED** [review]
LGTM. FYI @gspencergoog

**Comment by jmagman** [comment]
Google testing failing because https://github.com/flutter/flutter/pull/92861 and #92131 have not yet rolled.

**Pull request merged**